test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -375,67 +422,43 @@ void test_ucp_ep_reconfig::run(bool bidirectional)
     auto r_sender   = static_cast<const entity*>(&sender());
     auto r_receiver = static_cast<const entity*>(&receiver());
 
-    EXPECT_NE(r_sender->is_reconfigured(), r_receiver->is_reconfigured());
+    if (should_reconfigure()) {
+        EXPECT_NE(r_sender->is_reconfigured(), r_receiver->is_reconfigured());
+    } else {
+        EXPECT_FALSE(r_sender->is_reconfigured());
+        EXPECT_FALSE(r_receiver->is_reconfigured());
+    }
 
     r_sender->verify_configuration(*r_receiver, r_sender->num_reused_rscs());
     r_receiver->verify_configuration(*r_sender, r_sender->num_reused_rscs());
 }
 
-void test_ucp_ep_reconfig::skip_non_p2p()
-{
-    if (!has_p2p_transport()) {
-        UCS_TEST_SKIP_R("No p2p TLs available, config will be non-wireup");
-    }
-}
-
-void test_ucp_ep_reconfig::ensure_reused_lanes_reconfigurable()
+UCS_TEST_P(test_ucp_ep_reconfig, basic)
 {
-    if (!reuse_lanes()) {
-        return;
-    }
-
-    if (has_transport("tcp") || has_transport("dc_x") || has_transport("shm")) {
-        UCS_TEST_SKIP_R("non wired-up lanes are not supported yet");
+    if (has_transport("shm")) {
+        UCS_TEST_SKIP_R("TODO: add support for reconfiguration of separate "
+                        "wireup and AM lanes");
     }
-}
 
-/* TODO: Remove skip condition after next PRs are merged. */
-UCS_TEST_SKIP_COND_P(test_ucp_ep_reconfig, basic, !has_transport("rc"))
-{
     run();
 }
 
 UCS_TEST_P(test_ucp_ep_reconfig, request_reset, "PROTO_REQUEST_RESET=y")
 {
-    if (is_single_transport()) {
-        /* One side will consume all ifaces and the other side will have no ifaces left to use */
-        UCS_TEST_SKIP_R("exclude_iface requires at least 2 transports to work "
-                        "(for example DC + SHM)");
-    }
-
-    skip_non_p2p();
     run();
 }
 
-UCS_TEST_SKIP_COND_P(test_ucp_ep_reconfig, resolve_remote_id, is_self(),
-                     "MAX_RNDV_LANES=0")
+UCS_TEST_P(test_ucp_ep_reconfig, resolve_remote_id)
 {
-    if (has_transport("tcp")) {
-        UCS_TEST_SKIP_R("asymmetric setup is not supported for this transport "
-                        "due to a reachability issue - only matching "
-                        "interfaces can connect");
-    }
-
-    if (has_transport("shm")) {
-        UCS_TEST_SKIP_R("AM messages might be sent before reconfiguration"
-                        "(would be supported in next PR)");
+    if (has_transport("dc_x")) {
+        /* RDMA_READ between CX7 and BF devices over DC fails due to HW bug */

```

Comment:
why does reconfigure test expose the issue?

Reply:
because we split tl_bitmap between sender/receiver and thus try to connet mlx5_0 to mlx5_1